frontend/src/app/carousel/carousel.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, Input } from '@angular/core'
+import { trigger, state, style, transition, animate } from '@angular/animations';
+
+
+@Component({
+    selector: 'app-carousel',
+    templateUrl: './carousel.component.html',
+    styleUrls: ['./carousel.component.css'],
+    animations: [
+        trigger('imgState', [
+            state('inactive', style({
+                opacity: 0
+            })),
+            state('active', style({
+                opacity: 1
+            })),
+            transition('inactive => active', animate('300ms ease-in')),
+            transition('active => inactive', animate('300ms ease-in')),
+        ])
+    ]
+})
+
+export class CarouselComponent {
+
+    @Input() images: Array<string>;
+    counter = 0;
+
+    onClickInc() {
+        const total = this.images.length - 1;
+        this.counter = this.counter < total ? this.counter + 1 : 0;
+        this.setActive(`btn${this.counter}`);
+    }
+    onClickDec() {
+        const total = this.images.length - 1;
+        this.counter = this.counter > 0 ? this.counter - 1 : total;
+        this.setActive(`btn${this.counter}`);
+    }
+    onClickItem(event) {
+        this.counter = Number(event.explicitOriginalTarget.value);
+        this.setActive(`btn${this.counter}`);
+    }
+
+    setActive = function (buttonName) {
```

Comment:
Declare this method and method below in the same way as method onClickItem for consistency.